rust/validation/src/validation/list.rs

```diff
@@ -96,18 +89,10 @@ fn validate_unique_keys(schema: &List, items: &[Value], ctx: &mut Context) {
         for (current_trail, current_item) in &items {
             for (trail, item) in &items {
                 if current_trail != trail && current_item == item {
-                    ctx.violations.push(Feedback {
-                        path: {
-                            let mut path = ctx.path.clone();
-                            path.extend_from_slice(current_trail);
-                            path
-                        },
+                    ctx.result.errors.push(Feedback {
```

Comment:
I feel the compelxity of this method could be enhanced as right now we are going to loop over the items for each unique_key and then do a n square comparison - we could instead build a per unique key `Map <vlaue, list<trail>>`where we add the trail for each value (going through items only once) and then iter over the map and add a violation for any Map value